btk/create_observing_generator.py

```diff
@@ -35,69 +35,81 @@ def make_wcs(
     return w
 
 
-def default_obs_conditions(Args, band):
+def default_obs_conditions(survey_name, band):
     """Returns the default observing conditions from the WLD package
     for a given survey_name and band.
 
     Args:
-        Args: A `btk.config.SimulationParams` object containing parameters to generate blends
+        survey_name: Name of the survey which should be available in descwl
         band: filter name to get observing conditions for.
     Returns:
         `survey`: Dictionary containing the observing conditions and WCS information.
     """
     survey = descwl.survey.Survey.get_defaults(
-        survey_name=Args.survey_name, filter_band=band
+        survey_name=survey_name, filter_band=band
     )
     survey["center_sky"] = None
     survey["center_pix"] = None
     survey["projection"] = "TAN"
     return survey
 
 
-def generate(Args, obs_function=None):
-    """Generates class with observing conditions in each band.
+class ObservingGenerator:
+    def __init__(
+        self,
+        survey_name,
+        stamp_size,
+        obs_function=default_obs_conditions,
+        verbose=False,
+    ):
+        """Generates class with observing conditions in each band.
 
-    Args:
-        Args: Class containing input parameters.
-        obs_function: Function that outputs dict of observing conditions. If
-            not provided then the default `descwl.survey.Survey` values for the
-            corresponding Args.survey_name are used to create the
-            observing_generator.
+        Args:
+             obs_function: Function that outputs dict of observing conditions. If
+                           not provided, then the default `descwl.survey.Survey` values
+                           for the corresponding survey_name are used to create the
+                           observing_generator.
+             stamp_size: In arcseconds.
+        """
+        if survey_name not in btk.survey.surveys:
```

Comment:
May become an issue later if we use other image generation than the one from WLD. I am not sure whether it is actually necessary for WLD, if a custom obs_function creating a custom Survey object is used.

Reply:
Good point, I was thinking we can create different classes for other types of observing conditions that not necessarily return a `descwl.survey.Survey` object.